src/7fd10_len_b40.c

```diff
@@ -12,52 +24,149 @@ void func_800E6860(void) {
     }
 }
 
-// playerStatus getting coppied to a0 - how? inlining?
-#ifdef NON_EQUIVALENT
 s32 func_800E6904(void) {
     PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
-    //PlayerStatus* playerStatus = &gPlayerStatus; // doesn't change codegen to use it, so commented out for now
-    s32 actionState = gPlayerStatus.actionState;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 actionState = playerStatus->actionState;
 
-    if (!(gPlayerStatus.animFlags & 0x100000)) {
-        if (partnerActionStatus->actionState.b[0] == 0) {
-            if (!(gPlayerStatus.flags & 0x1000)) {
-                if (actionState == ACTION_STATE_IDLE || actionState == ACTION_STATE_WALK || actionState == ACTION_STATE_RUN) {
-                    return 1;
-                }
+    if (playerStatus->animFlags & PLAYER_STATUS_ANIM_FLAGS_100000) {
+        return FALSE;
+    }
+
+    if (partnerActionStatus->actionState.b[0] == 0) {
+        if (!(playerStatus->flags & PLAYER_STATUS_FLAGS_1000) &&
+            (actionState == ACTION_STATE_IDLE || actionState == ACTION_STATE_WALK || actionState == ACTION_STATE_RUN))
+        {
+            return TRUE;
+        }
+    } else if (partner_player_can_pause()) {
+        if (partnerActionStatus->actionState.b[3] == 6) {
+            return TRUE;
+        } else if (partnerActionStatus->actionState.b[3] == 9) {
+            if (actionState == ACTION_STATE_RIDE) {
+                return TRUE;
+            }
+        } else if (partnerActionStatus->actionState.b[3] == 8) {
+            if (actionState == ACTION_STATE_RIDE) {
+                return TRUE;
             }
-        } else if (partner_player_can_pause()) {
-            if (partnerActionStatus->actionState.b[3] == 6) {
-                return 1;
-            } else if (partnerActionStatus->actionState.b[3] == 9) {
-                if (actionState == ACTION_STATE_RIDE) {
-                    return 1;
+            sfx_play_sound(SOUND_MENU_ERROR);
+        }
+    }
+    return FALSE;
+}
+
+s32 can_pause(s32 currentButtons, s32 pressedButtons) {
```

Comment:
Should probably rename this to `try_pause` at some point given it plays sfx in some cases